xcpmd/src/vm-actions-module.c

```diff
@@ -707,9 +767,11 @@ void shutdown_vpnvm_dependencies_of_vm_by_name (struct arg_node * args) {
 void shutdown_vpnvm_dependencies_of_vm_by_uuid (struct arg_node * args) {
```

Comment:
So much code duplication!
Couldn't you create a couple of generic functions to avoid changing code in 22 places each time?